File: YABA-New/YABA/app/src/main/java/dev/subfly/yaba/core/model/annotation/AnnotationReadableCreateRequest.kt
package dev.subfly.yaba.core.model.annotation

import dev.subfly.yaba.core.model.utils.YabaColor

/**
 * Result payload for creating a READABLE annotation after the user confirms the sheet.
 * The owning screen applies the TipTap mark, persists JSON, then calls [dev.subfly.yabacore.managers.AnnotationManager].
 */
data class AnnotationReadableCreateRequest(
    val selectionDraft: ReadableSelectionDraft,
    val colorRole: YabaColor,
    val note: String?,
)

File: Darwin/YABACore/Model/Annotation/AnnotationReadableCreateRequest.swift
//
//  AnnotationReadableCreateRequest.swift
//  YABACore
//
//  Parity with Compose `AnnotationReadableCreateRequest`.
//

import Foundation

public struct AnnotationReadableCreateRequest: Sendable, Equatable {
    public var selectionDraft: ReadableSelectionDraft
    public var colorRole: YabaColor
    public var note: String?

    public init(
        selectionDraft: ReadableSelectionDraft,
        colorRole: YabaColor,
        note: String?
    ) {
        self.selectionDraft = selectionDraft
        self.colorRole = colorRole
        self.note = note
    }
}
